src/structures/context.ts

```diff
@@ -500,6 +577,37 @@ export class Context {
     );
   }
 
+  // /**
+  //  * Reply a message using button template
+  //  *
+  //  * @param {"basic" | "list" | "template"} type Type of button
+  //  * @param {AnyMessageContent?} options Reply message options
+  //  */
+  // public async replyWithButton<T extends ButtonType = 'basic'>(
+  // type: T,
+  // options: ButtonOptions<T>,
+  //   ) {
+  //     switch (type) {
+  //       case 'basic':
+  //         return this.sendRaw(options, { quoted: this.raw });
+  //       case 'list':
+  //         return this.sendRaw(options, { quoted: this.raw });
+  //       case 'template':
+  //         return this.sendRaw(options, { quoted: this.raw });
+  //       default:
+  //         return this.sendRaw({ viewOnce: true, ...options }, { quoted: this.raw });
+  //     }
+  //   }
+
+  /**
+   * Reply a message using button template
+   *
+   * @param {AnyMessageContent} options Reply message options
+   */
+  public async replyWithButton(options: AnyMessageContent) {
```

Comment:
I think, for `replyWithButton` or `replyWithButtons` method, we should accept the `buttons`, and `options` params.